cilium/cmd/cleanup.go

```diff
@@ -511,3 +533,47 @@ func removeTCFilters(linkAndFilters map[string][]*netlink.BpfFilter) error {
 
 	return nil
 }
+
+func removeXDPs(links []netlink.Link) error {
+	for _, link := range links {
+		err := netlink.LinkSetXdpFd(link, -1)
+		if err != nil {
+			return err
+		}
+		log.Infof("removed cilium xdp of %s", link.Attrs().Name)
+	}
+	return nil
+}
+
+func isLinkAttachedCiliumXDP(link netlink.Link) (bool, error) {
+	linkxdp := link.Attrs().Xdp
+	if linkxdp == nil || !linkxdp.Attached {
+		return false, nil
+	}
+
+	ok, err := isCiliumXDP(linkxdp.ProgId)
+	if ok && err == nil {
+		return true, nil
+	}
+
+	return false, err
+}
+
+func isCiliumXDP(progId uint32) (bool, error) {
+	xdp, err := ebpf.NewProgramFromID(ebpf.ProgramID(progId))
+	if err != nil {
+		return false, err
+	}
+
+	info, err := xdp.Info()
+	if err != nil {
+		return false, err
+	}
+
+	if strings.Contains(info.Name, "bpf_xdp_entry") {
```

Comment:
I am a bit paranoid that others might use the same entry point name. Mind renaming?

Reply:
Oh, I see https://github.com/cilium/cilium/pull/19735#discussion_r879625788. @ti-mo mind elaborating what's your concern regarding changing the function name? Also, cc @borkmann just in case I'm missing something.

Reply:
We currently don't prefix any bpf entrypoint function names with `cilium_`, so if we want to change it, we should change it across the whole codebase so it's consistent. A change like this should be discussed and planned, as we might for example, want to go with `cil_` (and remove the redundant `bpf_` prefix -- yeah, it's a bpf program alright!) to try to stay under the bpf_obj_name length limit as much as possible. See a similar discussion at https://github.com/cilium/cilium/issues/10067#issuecomment-1024322477. (cc @YutaroHayakawa @pchaigno)

Also, I started relying on the entrypoint name to classify the program type in my Go ELF loader patch: https://github.com/cilium/cilium/pull/19159/commits/f5db63ea6de9a51c31ca987e73630a8845fcc9ad#diff-d6f11688bc0f4227c8d78948312a1ca6e290c1c63dcf8c0db12a45d035fb9d5dR139-R140. We should switch to libbpf section name prefixes here to classify them automatically, but it's just an example of something that might break when changing this.

tl;dr: let's discuss this first, please

Reply:
Let's discuss this with the wider team at the sig-datapath meeting on Thu.

Reply:
I've added to this week's sig-datapath agenda to discuss the renaming https://docs.google.com/document/d/1IZZ-FTM_U3zN06-M_mDOgep_DcDM0Oe3YOFVM1BTIt8/edit#heading=h.f7lmu4rc837r.

Reply:
@zhanghe9702 We have discussed the issue, and decided to prefix the entry function with `cli_`. Perhaps you could rename it to `cli_xdp_entry` which should not exceed the 15 char limit.

Reply:
@brb I think you mean `cil_` and `cil_xdp_entry`.